src/app/ui/components/privacy/private-text.layout.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+import { type HTMLStyledProps, styled } from 'leather-styles/jsx';
+
+import { BasicTooltip } from '@app/ui/components/tooltip/basic-tooltip';
+
+interface PrivateTextLayoutProps extends HTMLStyledProps<'span'> {
+  children: React.ReactNode;
+  isPrivate?: boolean;
+  onShowValue?(): void;
+}
+
+export function PrivateTextLayout({
+  isPrivate,
+  onShowValue,
+  children,
+  ...rest
+}: PrivateTextLayoutProps) {
+  const canShowValue = isPrivate && onShowValue;
+
+  return (
+    <BasicTooltip label="Show value" disabled={!canShowValue} asChild>
+      <styled.span
+        {...rest}
+        onClick={canShowValue ? onShowValue : undefined}
+        cursor={canShowValue ? 'pointer' : undefined}
+        display="inline-block"
+        overflow="hidden"
+        fontFamily={isPrivate ? 'Diatype !important' : undefined}
```

Comment:
Maybe we can use one of [our typography tokens](https://github.com/leather-io/mono/blob/dev/packages/tokens/src/typography.ts) here for the font? Like `heading.01`?

If that's not working that's OK. I think it could be better to force it by using a `style={{` tag instead of `!important`:
```
 style={{ fontFamily:isPrivate ? 'Diatype' : undefined }}
```